pkg/addon/versioned_registry.go

```diff
@@ -131,7 +131,7 @@ func (i versionedRegistry) loadAddon(ctx context.Context, name, version string)
 		return nil, fmt.Errorf("specified version %s not exist", version)
 	}
 	for _, chartURL := range addonVersion.URLs {
-		archive, err := common.HTTPGet(ctx, chartURL)
+		archive, err := common.HTTPGetWithOption(ctx, chartURL, nil)
```

Comment:
why versioned_registry don't pass the arguments?

Reply:
Our addon repo is public just now, we can support private repo if user want this feature later.  And this is very easy, I think we can make a good first issue to let contributor to solve.

Reply:
This is not a good first issue, you should provide some design for this issue.

Reply:
Yes, will support this as a whole solution for user build a private addon repo with chartmuseum.